lib/index.ts

```diff
@@ -134,6 +165,87 @@ export default class VM extends AsyncEventEmitter {
     // We cache this promisified function as it's called from the main execution loop, and
     // promisifying each time has a huge performance impact.
     this._emit = promisify(this.emit.bind(this))
+
+    // TODO: Add comments here
+    logger.log('Setting up OVM contract objects')
+
+    this._emGasLimit = opts.emGasLimit || 100_000_000
+    this._initialized = opts.initialized || false
+    this._ovmStateManager = new OvmStateManager({ vm: this })
+
+    if (opts.contracts) {
+      this._contracts = opts.contracts
+    } else {
+      this._contracts.AddressResolver = makeOvmContract(this, 'AddressResolver')
+      this._contracts.ExecutionManager = makeOvmContract(this, 'ExecutionManager')
+      this._contracts.StateManager = makeOvmContract(this, 'StateManager')
+      this._contracts.SafetyChecker = makeOvmContract(this, 'SafetyChecker')
+    }
+  }
+
+  async init(): Promise<void> {
+    if (this._initialized) {
+      return
+    }
+    this._initialized = true
+
+    logger.log('Running OVM initialization logic')
+    await sleep(1000)
+
+    try {
+      // Contract deployment
+      logger.log(`Deploying OVM contracts:`)
+
+      logger.log(`Deploying AddressResolver...`)
+      await this._contracts.AddressResolver.deploy()
+      logger.log(`Deployed AddressResolver at: ${this._contracts.AddressResolver.addressHex}`)
+
+      logger.log(`Deploying StateManager...`)
+      await this._contracts.StateManager.deploy()
```

Comment:
Is this contract actually used in terms of code? Couldn't we just do `setAddress` since its calls are shimmed anyway?